src/ghci/stdout.rs

```diff
@@ -183,4 +201,33 @@ impl GhciStdout {
         let _ = sender.send(map);
         Ok(())
     }
+
+    #[instrument(skip(self), level = "debug")]
+    async fn set_mode(&mut self, sender: oneshot::Sender<()>, mode: Mode) {
+        self.mode = mode;
+        let _ = sender.send(());
+    }
+
+    async fn get_status_from_compile_output(&mut self, mut lines: Lines) -> miette::Result<()> {
+        // TODO: This would be pretty clumsy if we wanted to use this regex in multiple places.
+        // Might be worth bringing in `lazy_static`.
+        let compilation_finished_re = COMPILATION_FINISHED_RE
+            .get_or_init(|| Regex::new(r"^(?:Ok|Failed), [0-9]+ modules loaded.$").unwrap());
```

Comment:
Or you could use a function with a static.

```rust
fn compilation_finished_re() -> Regex {
    static RE: OnceCell<Regex> = OnceCell::new();
    RE.get_or_init(|| Regex::new(r"^(?:Ok|Failed), [0-9]+ modules loaded.$").unwrap())
}

```

Reply:
Good idea, implemented.